diff -U3 -Nr a/src/backend/main/main.c b/src/backend/main/main.c
--- a/src/backend/main/main.c
+++ b/src/backend/main/main.c
@@ -75,0 +76,4 @@
+#ifdef ORNA_EMBEDDED_ENTRY
+	int			clear_pglocaledir_result;
+	int			clear_pgsysconfdir_result;
+#endif
@@ -76,0 +81,8 @@
+#ifdef ORNA_EMBEDDED_ENTRY
+	clear_pgsysconfdir_result = unsetenv("PGSYSCONFDIR");
+	clear_pglocaledir_result = unsetenv("PGLOCALEDIR");
+	if (clear_pgsysconfdir_result != 0 ||
+		clear_pglocaledir_result != 0 ||
+		setlocale(LC_ALL, "C.UTF-8") == NULL)
+		return 1;
+#endif
@@ -80,3 +92,6 @@
-
-	reached_main = true;
-
+#ifdef ORNA_EMBEDDED_ENTRY
+	strlcpy(my_exec_path, "/usr/bin/orna", MAXPGPATH);
+#endif
+
+	reached_main = true;
+
@@ -127,5 +142,7 @@
-	/*
-	 * Set up locale information
-	 */
-	set_pglocale_pgservice(argv[0], PG_TEXTDOMAIN("postgres"));
-
+	/*
+	 * Set up locale information
+	 */
+#ifndef ORNA_EMBEDDED_ENTRY
+	set_pglocale_pgservice(argv[0], PG_TEXTDOMAIN("postgres"));
+#endif
+
diff -U0 -Nr a/src/bin/initdb/initdb.c b/src/bin/initdb/initdb.c
--- a/src/bin/initdb/initdb.c	2026-08-11 22:45:38.560588882 +0100
+++ b/src/bin/initdb/initdb.c	2026-08-11 22:45:51.881491448 +0100
@@ -3403,0 +3404 @@
+#ifndef ORNA_EMBEDDED_INITDB_ENTRY
@@ -3404,0 +3406 @@
+#endif
@@ -3750,0 +3753,2 @@
+	int			clear_pglocaledir_result;
+	int			clear_pgsysconfdir_result;
@@ -3752 +3756,6 @@
-	if (data_directory == NULL || data_directory[0] != '/' ||
+	clear_pgsysconfdir_result = unsetenv("PGSYSCONFDIR");
+	clear_pglocaledir_result = unsetenv("PGLOCALEDIR");
+	if (clear_pgsysconfdir_result != 0 ||
+		clear_pglocaledir_result != 0 ||
+		setlocale(LC_ALL, "C") == NULL ||
+		data_directory == NULL || data_directory[0] != '/' ||
